Backend/Routes/users.js

```diff
@@ -0,0 +1,85 @@
+import express from 'express';
+import bcrypt from 'bcrypt';
+import { User } from '../models/user.js';
+import { Op } from 'sequelize';
+import { PrismaClient } from '@prisma/client';
+
+const router = express.Router();
+
+// Route for user registration
+router.post('/users/signup', async (req, res) => {
+
+  const { name, password, email, userRole } = req.body;
+
+
+  try {
+    // Check if email already exists
+    const existingUser = await User.findOne({
+      where: {
+        [Op.or]: [{ email }]
+      }
+    });
+
+    if (existingUser) {
+      return res.status(400).json({ error: 'Email already exists' });
+    }
+
+    // Encrypt the password
+    const hashedPassword = await bcrypt.hash(password, 10);
```

Comment:
lets make 10 a constant value, that way it has a name, and if it needs to be referenced anywhere else, there is a single source of truth. 
In general any kind of "special number" should be a const

Reply:
Resolved